internal/rsl/rsl.go

```diff
@@ -174,6 +228,48 @@ func (a *AnnotationEntry) Commit(repo *gitinterface.Repository, sign bool) error
 	return err
 }
 
+// CommitUsingSpecificKey creates a commit object in the RSL for the
+// AnnotationEntry. The commit is signed using the provided PEM encoded SSH or
+// GPG private key. This is only intended for use in gittuf's developer mode or
+// in tests.
+func (a *AnnotationEntry) CommitUsingSpecificKey(repo *gitinterface.Repository, signingKeyBytes []byte) error {
+	// Check if referred entries exist in the RSL namespace.
+	for _, id := range a.RSLEntryIDs {
+		if _, err := GetEntry(repo, id); err != nil {
+			return err
+		}
+	}
+
+	latestEntry, err := GetLatestEntry(repo)
+	if err == nil {
+		a.Number = latestEntry.GetNumber() + 1
+	} else {
+		if errors.Is(err, ErrRSLEntryNotFound) {
+			// First entry -> can an annotation actually be first? TODO
+			a.Number = 1
+		} else {
+			return err
+		}
+	}
```

Comment:
This looks highly duplicated from ReferenceEntry and AnnotationEntry. Can we combine these at all?

I'm worried these may skew in unintended ways in the future.

Reply:
moved to helper to `setEntryNumber` appropriately